lib/babe/babe.go

```diff
@@ -243,22 +236,33 @@ func (b *Service) Resume() error {
 	b.Lock()
```

Comment:
Acquire lock after checking `b.IsPaused`

Reply:
I think this function should be atomic in the case that `Pause` or `Resume` are called concurrently by multiple processes, process A might see it's not paused, acquire the lock, then do `b.initiate()` but in between seeing it's not paused an acquiring the lock another process might also see the same thing, thus causing `b.initiate()` to be called twice which would be bad